conda/build_pytorch.sh

```diff
@@ -388,7 +388,18 @@ for py_ver in "${DESIRED_PYTHON[@]}"; do
 
     # Install the built package and run tests, unless it's for mac cross compiled arm64
     if [[ -z "$CROSS_COMPILE_ARM64" ]]; then
-        conda install -y "$built_package"
+        # Install the package as if from local repo instead of tar.bz2 directly in order
+        # to trigger runtime dependency installation. See https://github.com/conda/conda/issues/1884
+        # Notes:
+        # - pytorch-nightly is included to install torchtriton
+        # - nvidia is included for cuda builds, there's no harm in listing the channel for cpu builds
+        if [[ "$OSTYPE" == "msys" ]]; then
+          # note the extra slash: `pwd -W` returns `c:/path/to/dir`, we need to add an extra slash for the URI
+          local_channel="/$(pwd -W)/$output_folder"
+        else
+          local_channel="$(pwd)/$output_folder"
+        fi
+        conda install -y -c "file://$local_channel" pytorch==$PYTORCH_BUILD_VERSION -c pytorch -c numba/label/dev -c pytorch-nightly -c nvidia
```

Comment:
We should also include pytorch-cuda="XX.X" if we are testing for CUDA

Reply:
It's not needed, the package we built already has the right CUDA dep. E.g. I just checked one of them and it has `    "pytorch-cuda >=11.7,<11.8"` in the conda's spec